File: exemplos/js/07_categorias.mjs
// Vocabulário de categorias do inquilino (GET /api/categorias, escopo catalogo:ler)
import { chamar, exigir, exigirChave } from "./_apoio.mjs";

exigirChave();
const { status, objeto } = await chamar("/api/categorias");
exigir(status === 200, `esperava 200 em /api/categorias, veio ${status}: ${JSON.stringify(objeto)}`);
const lista = Array.isArray(objeto) ? objeto : (objeto.categorias ?? []);
console.log(`${lista.length} categoria(s)`);
for (const c of lista.slice(0, 20)) console.log("  " + (typeof c === "object" ? c.nome : c));

File: exemplos/python/07_categorias.py
# Vocabulário de categorias do inquilino (GET /api/categorias, escopo catalogo:ler)
from _apoio import chamar, exigir, exigir_chave

exigir_chave()
status, _cab, corpo = chamar("/api/categorias")
exigir(status == 200, f"esperava 200 em /api/categorias, veio {status}: {corpo}")
lista = corpo if isinstance(corpo, list) else corpo.get("categorias", [])
print(f"{len(lista)} categoria(s)")
for c in lista[:20]:
    print("  " + (c.get("nome") if isinstance(c, dict) else str(c)))
